capsul/study_config/config_modules/mrtrix_config.py

```diff
@@ -0,0 +1,123 @@
+        self.study_config.add_trait('mrtrix_path', File(
+            Undefined,
+            output=False,
+            desc='Parameter to specify the mrtrix path', groups=['mrtrix']))
+        self.study_config.add_trait('use_mrtrix', Bool(
+            Undefined,
+            output=False,
+            desc='Parameter to tell that we need to configure mrtrix',
+            groups=['mrtrix']))
+
+    def __del__(self):
+        try:
+            self.study_config.engine.settings.module_notifiers.get(
+                'capsul.engine.module.mrtrix', []).remove(
+                    self.sync_from_engine)
+        except (ValueError, ReferenceError):
+            pass
+
+    def initialize_module(self):
+        """ Set up mrtrix environment variables according to current
+        configuration.
+        """
+        if 'capsul.engine.module.mrtrix' \
+                not in self.study_config.engine._loaded_modules:
+            self.study_config.engine.load_module('capsul.engine.module.mrtrix')
+        if type(self.study_config.engine) is not CapsulEngine:
+            # engine is a proxy, thus we are initialized from a real
+            # CapsulEngine, which holds the reference values
+            self.sync_from_engine()
+        else:
+            self.sync_to_engine()
+        # TODO : do we need to add this tets for mrtrix as for afni/ants/fsl ?
+        # # this test aims to raise an exception in case of incorrect setting,
+        # # complying to capsul 2.x behavior.
+        # if self.study_config.use_mrtrix is True:
+        #     check_mrtrix_configuration(self.study_config)
+
+    def initialize_callbacks(self):
+        self.study_config.on_trait_change(
+            self.sync_to_engine, '[mrtrix_path, use_mrtrix]')
+        #  WARNING ref to self in callback
+        self.study_config.engine.settings.module_notifiers[
+            'capsul.engine.module.mrtrix'] = [self.sync_from_engine]
+
+    def sync_to_engine(self, param=None, value=None):
+        if getattr(self, '_syncing', False):
+            # manage recursive calls
+            return
+        self._syncing = True
+        try:
+            if 'MRTRIXConfig' in self.study_config.modules \
+                    and 'capsul.engine.module.mrtrix' \
+                        in self.study_config.engine._loaded_modules:
+                with self.study_config.engine.settings as session:
+                    cif = self.study_config.engine.settings.config_id_field
+                    config = session.config('mrtrix', 'global')
+                    mrtrix_path = self.study_config.mrtrix_path
+                    if mrtrix_path is Undefined:
+                        mrtrix_path = None
+
+                    if config is None:
+                        session.new_config(
+                            'mrtrix', 'global',
+                            {'directory': mrtrix_path,
+                             cif: 'mrtrix'})
```

Comment:
I must admit here that I do not know how cif values are used. In a previous comment I supposed it was automatically generated with a random value if not set, but I do not know why there is a fixed value here. @denisri might have the answer.

Reply:
I don't remember if the CIF is automatically generated if not specified, I guess yes but I have not checked (just coming back from vacation...). It is used as a key in the config database so it just has to be unique in the config CIF values. When several configs for the same software may be configured, the CIF must differ, that's why we use values like "spm12-standalone" or "spm8" to distinguish them, but it's just a convention.